### PR TITLE
Add support for Whitelist of Non-RFC-1918 Private Networks and MySQL Activities Logging

### PR DESCRIPTION
**Pull request by beibeiyang**
Some non-standard private networks require extra configuration so that internal file storage (WebDAV) can communicate with other PCF processes. 

This pull request adds support for `$blobstore_internal_access_rules` so that users can customize their whitelist of Non-RFC-1918 Private Networks for PCF Elastic Runtime over GCP and vSphere. 

The `blobstore_internal_access_rules` is given the default value `"allow 10.0.0.0/8;,allow 172.16.0.0/12;,allow 192.168.0.0/16;"`. But if users have a Non-RFC-1918 Private Network (such as `172.20.8.0/21`), they can define the following in params.yml for GCP or vSphere:

`blobstore_internal_access_rules: "allow 10.0.0.0/8;,allow 172.20.8.0/21;,allow 172.16.0.0/12;,allow 192.168.0.0/16;"`

This corresponds to the GUI config: Pivotal Elastic Runtime tile -> Advanced Features -> Whitelist for non-RFC-1918 Private Networks.

This pull request also adds support for `mysql_activity_logging`. If you want to log audit events for internal MySQL, select Enable server activity logging under Server Activity Logging. This flag is enabled by default in Elastic Runtime GUI.

This corresponds to the GUI config: Pivotal Elastic Runtime tile -> Internal MySQL -> Server Activity Logging

_Tested in vSphere ESXi6 with PCF 1.10 and Concourse 3.1.1 and 3.3.0_

Reference:
https://docs.pivotal.io/pivotalcf/latest/customizing/config-er-vmware.html#er-advanced
https://docs.pivotal.io/pivotalcf/1-10/customizing/config-er-vmware.html#internal-mysql

### CODE REVIEW

File: tasks/config-ert/load_cf_properties.sh [hunk 1]
```diff
@@ -19,6 +19,8 @@ jq \
   --arg disable_insecure_cookies "$DISABLE_INSECURE_COOKIES" \
   --arg router_request_timeout_seconds "$ROUTER_REQUEST_TIMEOUT_IN_SEC" \
   --arg mysql_monitor_email "$MYSQL_MONITOR_EMAIL" \
+  --arg garden_network_pool "$GARDEN_NETWORK_POOL_CIDR" \
```
Comment (krishicks) [inline tasks/config-ert/load_cf_properties.sh]: This needs to be removed as it is no longer supported in ERT 1.11.

File: tasks/config-ert/load_cf_properties.sh [hunk 2]
```diff
@@ -19,6 +19,8 @@ jq \
   --arg disable_insecure_cookies "$DISABLE_INSECURE_COOKIES" \
   --arg router_request_timeout_seconds "$ROUTER_REQUEST_TIMEOUT_IN_SEC" \
   --arg mysql_monitor_email "$MYSQL_MONITOR_EMAIL" \
+  --arg garden_network_pool "$GARDEN_NETWORK_POOL_CIDR" \    
+  --arg garden_network_mtu "$GARDEN_NETWORK_MTU" \
```
Comment (krishicks) [inline tasks/config-ert/load_cf_properties.sh]: This needs to be removed as it is no longer supported in ERT 1.11.

File: tasks/config-ert/load_cf_properties.sh [hunk 3]
```diff
@@ -110,6 +113,12 @@ jq \
     ".router.request_timeout_in_seconds": {
       "value": $router_request_timeout_seconds
     },
+    ".diego_cell.garden_network_pool": {    
+      "value": $garden_network_pool   
+    },    
+    ".diego_cell.garden_network_mtu": {   
+      "value": $garden_network_mtu    
+    },
```
Comment (krishicks) [inline tasks/config-ert/load_cf_properties.sh]: This needs to be removed as it is no longer supported in ERT 1.11.

File: tasks/config-ert/load_cf_properties.sh [hunk 4]
```diff
@@ -286,6 +295,19 @@ jq \
 
   +
 
+  # Whitelist for non-RFC-1918 Private Networks
+  if $blobstore_internal_access_rules != "" then
+    {
+      ".nfs_server.blobstore_internal_access_rules": {
+      "value": $blobstore_internal_access_rules
```
Comment (krishicks) [inline tasks/config-ert/load_cf_properties.sh]: This should be indented two spaces.

File: tasks/config-ert/task.sh [hunk 5]
```diff
@@ -73,8 +73,6 @@ cf_resources=$(
     "nats": $NATS_INSTANCES,
     "nfs_server": $NFS_SERVER_INSTANCES,
     "router": $ROUTER_INSTANCES,
-    "syslog_adapter": $SYSLOG_ADAPTER_INSTANCES,
-    "syslog_scheduler": $SYSLOG_SCHEDULER_INSTANCES,
```
Comment (krishicks) [inline tasks/config-ert/task.sh]: This needs to be restored as it is required by ERT 1.11.

File: tasks/config-ert/task.yml [hunk 6]
```diff
@@ -96,8 +97,6 @@ params:
   DOPPLER_INSTANCES:
   LOGGREGATOR_TC_INSTANCES:
   TCP_ROUTER_INSTANCES:
-  SYSLOG_ADAPTER_INSTANCES:
-  SYSLOG_SCHEDULER_INSTANCES:
```
Comment (krishicks) [inline tasks/config-ert/task.yml]: This needs to be restored as it is required by ERT 1.11.

File: install-pcf/gcp/pipeline.yml [hunk 7]
```diff
@@ -276,6 +276,7 @@ jobs:
       db_silk_password: {{db_silk_password}}
       db_locket_username: {{db_locket_username}}
       db_locket_password: {{db_locket_password}}
+      BLOBSTORE_INTERNAL_ACCESS_RULES: {{blobstore_internal_access_rules}}
```
Comment (krishicks) [inline install-pcf/gcp/pipeline.yml]: This param also needs to be added to `pcf-pipelines/tasks/install-ert/configure-json/task.yml`, and used in `pcf-pipelines/tasks/install-ert/configure-json/task.sh`. Alternatively, remove this partly-implemented feature from the gcp and aws pipelines as only the vsphere pipeline fully implements it.

File: install-pcf/vsphere/params.yml [hunk 8]
```diff
@@ -171,7 +174,8 @@ router_static_ips:      # Comma-separated list of static IPs
 tcp_router_static_ips:  # Comma-separated list of static IPs
 ssh_static_ips:         # Comma-separated list of static IPs
 
-mysql_monitor_email:    # Email address to receive MySQL monitor notifications
+mysql_monitor_email:     # Email address to receive MySQL monitor notifications
+mysql_activity_logging: enable  # Whether to enable internal MySQL server activity logging (enable|disable)
```
Comment (beibeiyang) [inline install-pcf/vsphere/params.yml]: I have added support of `mysql_activity_logging`. This flag is enabled by default in Elastic Runtime GUI. But I kept running into an error `'mysql_monitor/0 (f4a5b5fb-0868-428a-94c6-6b4322aa5821)' is not running after update. Review logs for failed jobs: replication-canary` in vsphere installs. This error went away after I set `mysql_activity_logging: disable`

File: install-pcf/gcp/pipeline.yml (hunk 7 above)
Comment (beibeiyang) [inline install-pcf/gcp/pipeline.yml]: I have remove BLOBSTORE_INTERNAL_ACCESS_RULES in GCP. This is not implemented at all in AWS.

File: install-pcf/vsphere/params.yml (hunk 8 above)
Comment (krishicks) [inline install-pcf/vsphere/params.yml]: Please split this out from the PR; each PR should be for one feature.

File: tasks/config-ert/task.sh [hunk 9]
```diff
@@ -73,7 +73,7 @@ cf_resources=$(
     "nats": $NATS_INSTANCES,
     "nfs_server": $NFS_SERVER_INSTANCES,
     "router": $ROUTER_INSTANCES,
-    "syslog_adapter": $SYSLOG_ADAPTER_INSTANCES,
+    "syslog_adapter": $SYSLOG_ADAPTER_INSTANCES,
```
Comment (krishicks) [inline tasks/config-ert/task.sh]: Please remove this added whitespace.

File: tasks/config-ert/task.yml [hunk 10]
```diff
@@ -55,9 +56,8 @@ params:
   DISABLE_INSECURE_COOKIES:
   ROUTER_STATIC_IPS:
   ROUTER_REQUEST_TIMEOUT_IN_SEC:
-  GARDEN_NETWORK_POOL_CIDR:
-  GARDEN_NETWORK_MTU:
```
Comment (krishicks) [inline tasks/config-ert/task.yml]: Please rebase to remove this; it's been gone in master as of 9843501.

File: tasks/config-ert/task.yml [hunk 11]
```diff
@@ -96,7 +96,7 @@ params:
   DOPPLER_INSTANCES:
   LOGGREGATOR_TC_INSTANCES:
   TCP_ROUTER_INSTANCES:
-  SYSLOG_ADAPTER_INSTANCES:
+  SYSLOG_ADAPTER_INSTANCES:
```
Comment (krishicks) [inline tasks/config-ert/task.yml]: Please remove this added whitespace.

File: install-pcf/vsphere/params.yml [hunk 12]
```diff
@@ -152,6 +152,9 @@ mail_attribute_name:
 first_name_attribute:
 last_name_attribute:
 
+#Whitelist for non-RFC-1918 Private Networks: https://docs.pivotal.io/pivotalcf/latest/customizing/config-er-vmware.html#er-advanced
+blobstore_internal_access_rules: "allow 10.0.0.0/8;,allow 172.16.0.0/12;,allow 192.168.0.0/16;"
```
Comment (krishicks) [inline install-pcf/vsphere/params.yml]: This cannot be defaulted to be populated. Per the docs linked:

> Most PCF deployments do not require any modifications to this field.

**Comment by pivotal-issuemaster** [comment]
@beibeiyang Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/pivotal)!

[Click here](https://cla.pivotal.io/sync/pivotal?repositoryId=pivotal-cf/pcf-pipelines&pullRequestId=135) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by beibeiyang** [comment]
@abbyachau Yes, please add it as a feature request. Thank you.

**Comment by abbyachau** [comment]
Hi @beibeiyang I've created a tracker [story](https://www.pivotaltracker.com/story/show/154427150) for this feature request. Are you happy to close this pull request in favour of the tracker story? Or would you prefer we open a github issue to track this request?  I've also reached out to you via email to ask for more information about customer use cases so that I can prioritise this story accordingly. Thanks again.

**Comment by beibeiyang** [comment]
@abbyachau Thanks. I will close this pull request.

**Pull request closed without merging**